### ISSUE TITLE
Cannot reference the type IResult in a non web project

### ISSUE THREAD

**Issue by jxwaters** [dotnet/aspnetcore#45637]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I have a class library that needs to convert various exceptions into the new IResult used by Minimal APIs.
For instance, min apis can return Ok, which is found in Microsoft.AspnetCore.Http.TypedResults and is of type Ok, which is IResult.

`return TypedResults.Ok(string.Empty);`

My extension method looks like this

```
        public static IResult ToIResult<T>(this Result<T> val)
        {
            if (val.IsSuccess)
            {
                return TypedResults.Ok(val.Value);
            }

            switch (val.ErrorException)
            {
                case ValidationException ve:
                    var sb = new StringBuilder();
                    sb.Append(ve.Message);
                    sb.Append("\n");
                    foreach (var error in ve.Errors)
                    {
                        sb.Append($"  {error.Key}");
                        sb.Append("\n");
                        var vals = string.Join("\n", error.Value);
                        sb.Append(vals);
                    }
                    return TypedResults.BadRequest(sb.ToString());
                case NotFoundException nfe:
                    return TypedResults.NotFound(nfe.Message);
                case { } e:
                    throw e;
            }

             return TypedResults.Ok(string.Empty);
        }

```

The method body compiles, but the compiler cannot find the type IResult.

I can't for the life of me figure out which nuget to reference to resolve it.

Find this type  on nuget returns Microsoft.AspNetCore.App.Ref -but I cannot add that to my class library project, which is of type 

<Project Sdk="Microsoft.NET.Sdk">

What to do?

### .NET Version

7

**Comment by davidfowl** [comment]
https://learn.microsoft.com/en-us/aspnet/core/fundamentals/target-aspnetcore?view=aspnetcore-7.0&tabs=visual-studio

TL;DR Add:

```xml
<ItemGroup>
    <FrameworkReference Include="Microsoft.AspNetCore.App" />
</ItemGroup>
```

To your csproj

**Comment by davidfowl** [comment]
Also for `ValidationException` consider using `Results.ProblemDetails`
